pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,236 @@
+    Additionally, it also contains two more attributes (``register`` and ``pulses``) that contain
+    the information that the hardware needs to execute this Hamiltonian.
+
+    .. warning::
+
+        This class should NEVER be initialized directly! Please use the functions
+        :func:`rydberg_interaction` and :func:`rydberg_drive` instead.
+
+    .. seealso:: :func:`rydberg_interaction`, :func:`rydberg_drive`, :class:`ParametrizedHamiltonian`
+
+    Args:
+        coeffs (Union[float, callable]): coefficients of the Hamiltonian expression, which may be
+            constants or parametrized functions. All functions passed as ``coeffs`` must have two
+            arguments, the first one being the trainable parameters and the second one being time.
+        observables (Iterable[Observable]): observables in the Hamiltonian expression, of same
+            length as ``coeffs``
+
+    Keyword Args:
+        register (list): list of coordinates (in micrometers) of each atom in the ensemble
+        pulses (list): list of ``RydbergPulse`` classes containing the information about the
+            amplitude, phase, detuning and wires of each pulse
+        interaction_coeff (float): Rydberg interaction constant in units: :math:`MHz \times \mu m^6`.
+            Defaults to :math:`862690 \times 2\pi MHz \times \mu m^6`.
+
+    Returns:
+        RydbergHamiltonian: class representing the Hamiltonian of an ensemble of Rydberg atoms
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        coeffs,
+        observables,
+        register: list = None,
+        pulses: List["RydbergPulse"] = None,
+        interaction_coeff: float = 862690 * 2 * np.pi,
+    ):
+        self.register = register
+        self.pulses = [] if pulses is None else pulses
+        self.interaction_coeff = interaction_coeff
+        super().__init__(coeffs, observables)
+
+    def __add__(self, other):
+        if not isinstance(other, RydbergHamiltonian):
+            return super().__add__(other)
+
+        # Update coeffs, obs and hardware attributes
+        if self.register is not None:
+            if other.register is not None:
+                raise ValueError("We cannot add two Hamiltonians with an interaction term!")
+            if not self.wires.contains_wires(other.wires):
+                warnings.warn(
+                    "The wires of the laser fields are not present in the Rydberg ensemble."
+                )
+        elif other.register is not None and not other.wires.contains_wires(self.wires):
+            warnings.warn("The wires of the laser fields are not present in the Rydberg ensemble.")
+
+        new_register = self.register or other.register
+        new_pulses = self.pulses + other.pulses
+        new_ops = self.ops + other.ops
+        new_coeffs = self.coeffs + other.coeffs
+        return RydbergHamiltonian(new_coeffs, new_ops, register=new_register, pulses=new_pulses)
+
+
+@dataclass
+class RydbergPulse:
```

Comment:
Will this be not shown in the docs?

Reply:
It shouldn't! This is an internal feature used only to structure the pulse data